src/game.cpp

```diff
@@ -319,11 +320,18 @@ void game::load_core_data()
     // core data can be loaded only once and must be first
     // anyway.
     DynamicDataLoader::get_instance().unload_data();
+    MonsterGenerator::generator().reset(); // @todo Separate clearing from destroying. It's not the best place for keeping it.
```

Comment:
Regarding the comment `reset` here resets the content, it does not destruct the generator. This is already separate: destruction happens when the static instance in `MonsterGenerator::generator` is destructed at the end of the program.

`reset` should be called from `unload_data` there is no reason to move this out of there. But `unload_data` should not be called from `game::~game`.